### PR TITLE
Update dependencies for v1.13.0-rc.6 release

### PR DESCRIPTION
**Pull request by JoshVanL**

### CODE REVIEW

File: tests/e2e/upgrade/upgrade_test.go [hunk 1]
```diff
@@ -81,7 +81,39 @@ var supportedUpgradePaths = []upgradePath{
...
 	// test downgrade.
+	{
+		previous: common.VersionDetails{
+			RuntimeVersion:      "1.13.0-rc.1",
+			DashboardVersion:    "0.14.0",
+			ClusterRoles:        []string{"dapr-dashboard", "dapr-injector", "dapr-operator-admin", "dapr-placement", "dapr-sentry"},
+			ClusterRoleBindings: []string{"dapr-operator-admin", "dapr-dashboard", "dapr-injector", "dapr-placement", "dapr-sentry"},
+			CustomResourceDefs:  []string{"components.dapr.io", "configurations.dapr.io", "subscriptions.dapr.io", "resiliencies.dapr.io", "httpendpoints.dapr.io"},
+		},
+		previous: common.VersionDetails{
```
Comment (mukundansundar) [inline tests/e2e/upgrade/upgrade_test.go:109]:
```suggestion
		next: common.VersionDetails{
```

**Comment by JoshVanL** [comment]
Thanks for the patch fix @mukundansundar

**Comment by JoshVanL** [comment]
Marking draft until https://github.com/dapr/go-sdk/pull/503 is merged into main.

File: tests/e2e/upgrade/upgrade_test.go [hunk 2]
```diff
@@ -81,7 +81,39 @@ var supportedUpgradePaths = []upgradePath{
...
+	{
+		previous: common.VersionDetails{
+			RuntimeVersion:      "1.12.0",
+			DashboardVersion:    "0.14.0",
+			ClusterRoles:        []string{"dapr-dashboard", "dapr-injector", "dapr-operator-admin", "dapr-placement", "dapr-sentry"},
+			ClusterRoleBindings: []string{"dapr-operator-admin", "dapr-dashboard", "dapr-injector", "dapr-placement", "dapr-sentry"},
+			CustomResourceDefs:  []string{"components.dapr.io", "configurations.dapr.io", "subscriptions.dapr.io", "resiliencies.dapr.io", "httpendpoints.dapr.io"},
+		},
+		next: common.VersionDetails{
+			RuntimeVersion:      "1.13.0-rc.0",
```
Comment (mukundansundar) [inline tests/e2e/upgrade/upgrade_test.go:93]:
```suggestion
			RuntimeVersion:      "1.13.0-rc.1",
```

File: go.mod [hunk 3]
```diff
@@ -202,58 +209,49 @@ require (
...
 )
 
-replace (
-	github.com/Azure/go-autorest => github.com/Azure/go-autorest v14.2.0+incompatible
-	github.com/docker/docker => github.com/moby/moby v17.12.0-ce-rc1.0.20200618181300-9dc6525e6118+incompatible
-	github.com/russross/blackfriday => github.com/russross/blackfriday v1.5.2
-	k8s.io/cli-runtime => k8s.io/cli-runtime v0.25.2
-	k8s.io/client => github.com/kubernetes-client/go v0.0.0-20190928040339-c757968c4c36
-	k8s.io/client-go => k8s.io/client-go v0.25.2
-)
```
Comment (mukundansundar) [inline go.mod:259]: why are these removed?

File: go.mod [hunk 4]
```diff
@@ -202,58 +209,49 @@ require (
...
+replace github.com/dapr/go-sdk => github.com/joshvanl/go-sdk v0.0.0-20240206235854-ec87d533b50a
```
Comment (mukundansundar) [inline go.mod:257]: why is this replace needed?

**Review by mukundansundar — CHANGES_REQUESTED** [review]
the go-sdk replace needs to be removed.

File: go.mod (hunk 3 above)
Comment (JoshVanL) [reply go.mod:259]: Why were they required originally?

File: go.mod (hunk 4 above)
Comment (JoshVanL) [reply go.mod:257]: Pointing to my fork until it is merged into main.

File: go.mod (hunk 3 above)
Comment (mukundansundar) [reply go.mod:259]: I believe that these were added initially to maintain compatibility with docker, go-autorest (azure auth), and k8s client for (azure, gcp etc. auth)
cc @yaron2

Comment (yaron2) [reply go.mod:259]: Historical reasons of incompatibility. If the CLI compiles and tests pass there's no reason to bring them back

File: go.mod (hunk 4 above)
Comment (mukundansundar) [reply go.mod:257]: if this can at least point to go-sdk master branch commit or RC it would be great

Comment (cicoyle) [reply go.mod:257]: this is merged now. the replace can be rm-ed

File: go.mod (hunk 3 above)
Comment (mukundansundar) [reply go.mod:259]: the one suggestion is to keep go auto rest and k8s version as it is now for this release alone ...It can be changed immediately in the next release.

**Review by mukundansundar — CHANGES_REQUESTED** [review]
small change to 1.13.0-rc.2 for runtime version

**Review by yaron2 — APPROVED** [review]
